File: carelog-be/src/main/java/carelog/carelog/common/config/TenantContext.java
package carelog.carelog.common.config;

import java.util.UUID;

public class TenantContext {
    private static final ThreadLocal<UUID> ORGANIZATION_ID = new ThreadLocal<>();

    public static void set(UUID organizationId) { ORGANIZATION_ID.set(organizationId); }
    public static UUID get() { return ORGANIZATION_ID.get(); }
    public static void clear() { ORGANIZATION_ID.remove(); }
}

File: src/main/kotlin/carelog/carelog/common/config/TenantContext.kt
package carelog.carelog.common.config

import java.util.UUID

object TenantContext {
    private val organizationId = ThreadLocal<UUID?>()

    fun set(id: UUID?) = organizationId.set(id)
    fun get(): UUID? = organizationId.get()
    fun clear() = organizationId.remove()
}